File: OsmAnd/src/net/osmand/plus/settings/enums/WeightUnits.java
package net.osmand.plus.settings.enums;

import static net.osmand.shared.util.OsmAndFormatter.KILOGRAMS_IN_ONE_TON;
import static net.osmand.shared.util.OsmAndFormatter.POUNDS_IN_ONE_TON;

import androidx.annotation.NonNull;
import androidx.annotation.StringRes;

import net.osmand.plus.R;
import net.osmand.util.CollectionUtils;

import java.text.DecimalFormat;

public enum WeightUnits implements MeasurementUnits {

	TONES(R.string.shared_string_tones, R.string.metric_ton, 1),
	KILOGRAMS(R.string.shared_string_kilograms, R.string.kg, KILOGRAMS_IN_ONE_TON),
	POUNDS(R.string.shared_string_pounds, R.string.metric_lbs, POUNDS_IN_ONE_TON);

	private final int symbolResId;
	private final int titleResId;
	private final float coefficient;

	WeightUnits(@StringRes int titleResId, @StringRes int symbolResId, float coefficient) {
		this.titleResId = titleResId;
		this.symbolResId = symbolResId;
		this.coefficient = coefficient;
	}

	@StringRes
	@Override
	public int getNameResId() {
		return titleResId;
	}

	@StringRes
	@Override
	public int getSymbolResId() {
		return symbolResId;
	}

	@Override
	public float getConversionCoefficient() {
		return coefficient;
	}

	@Override
	public boolean isMetric() {
		return CollectionUtils.equalsToAny(this, TONES, KILOGRAMS);
	}

	@NonNull
	@Override
	public String formatValue(float value) {
		if (this == POUNDS) {
			DecimalFormat formatter = new DecimalFormat("#,###.#");
			return formatter.format(value);
		}
		return MeasurementUnits.super.formatValue(value);
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/units/WeightUnits.kt
package net.osmand.shared.units

object WeightConstants {
	const val KILOGRAMS_IN_TON = 1000.0
	const val POUNDS_IN_TON = 2204.622621848776
}

enum class WeightUnits(
	override val nameKey: String,
	override val symbolKey: String,
	override val conversionCoefficient: Double,
	override val system: MeasurementSystem
) : MeasurementUnit<WeightUnits> {

	TONS(
		nameKey = "shared_string_tones",
		symbolKey = "metric_ton",
		conversionCoefficient = 1.0,
		system = MeasurementSystem.METRIC
	),

	KILOGRAMS(
		nameKey = "shared_string_kilograms",
		symbolKey = "kg",
		conversionCoefficient = WeightConstants.KILOGRAMS_IN_TON,
		system = MeasurementSystem.METRIC
	),

	POUNDS(
		nameKey = "shared_string_pounds",
		symbolKey = "metric_lbs",
		conversionCoefficient = WeightConstants.POUNDS_IN_TON,
		system = MeasurementSystem.IMPERIAL
	);
}